cirq/protocols/decompose.py

```diff
@@ -198,11 +200,18 @@ def decompose(
             "not possible to get stuck if you don't have a criteria on what's "
             "acceptable to keep.")
 
-    if intercepting_decomposer is None:
-        decomposer = _default_decomposer
-    else:
-        decomposers = [intercepting_decomposer, _default_decomposer]
-        decomposer = lambda op: _intercept_decompose(op, decomposers)
+    decomposers = [d
```

Comment:
boy it would be nice to use a filter here instead of this funky looking list comprehension, but I guess that won't work in 2.7

Reply:
`functools` exists in both, I think. But eh, people like list comprehensions more than `map`/`filter` for whatever reason.